Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs

```diff
@@ -232,13 +71,14 @@ tfFulfillments res = do
           }
       ]
   where
-    mkFulfillmentState quoteType =
+    mkFulfillmentState quoteType = do
+      fulfilState <- bookingStatusCode quoteType
       Just $
         Spec.FulfillmentState
           { fulfillmentStateDescriptor =
               Just $
                 Spec.Descriptor
-                  { descriptorCode = Just (bookingStatusCode quoteType),
+                  { descriptorCode = Just fulfilState,
```

Comment:
You are sending Nothing and putting Just also.

Reply:
No if you check above part I have used a bind operator so if the `fulfillState` is `Nothing`. It will send `Nothing` for whole `FulFillmenState` as `Nothing`. else it will use it in the below way.